tests/test_rfree.py

```diff
@@ -0,0 +1,35 @@
+from rsbooster.utils import rfree
+import pytest
+import numpy as np
+
+
+def test_rfree():
+    """
+    Test that "seed" parameter creates reproducible rfree flags, whereas omitting it does not
+    """
+
+    params = {
+        "cell": (30, 40, 50, 90, 90, 90),
+        "sg": 19,
+        "dmin": 1.5,
+        "rfraction": 0.05,
+    }
+
+    flags = rfree(**params, seed=2022)
+    flags_same_seed = rfree(**params, seed=2022)
+    flags_different_seed = rfree(**params, seed=None)
+
+    assert all(flags == flags_same_seed)
+    assert any(flags != flags_different_seed)
+
+    return
+
+
+@pytest.mark.parametrize("rfraction", [0.01, 0.05, 0.1, 0.2])
+def test_rfree_fraction(rfraction, cell=(30, 40, 90, 90, 90, 90), sg=19, dmin=1.2, seed=None):
+
+    flags = rfree(cell=cell, sg=sg, dmin=dmin, rfraction=rfraction, seed=seed)
+
+    assert np.isclose(np.mean(flags["R-free-flags"].to_numpy()), rfraction, rtol=0.1)
```

Comment:
I was surprised (and a little concerned) that I had to be so generous with this tolerance parameter. With `rtol=0.05`, the test would fail about half the time (I chose to not seed this test, because ideally we have a test that always works.). Does this mean I've done something wrong? Does this have to do with using a 100% complete dataset?

Reply:
I'll look into this a bit more -- not exactly sure about the `rtol`, but I agree this seems like a high value

Reply:
So the `rtol=0.05` runs into problems with `rfraction=0.01`. In general, I think this reflects that we are using $Uniform(0, 1) \leq rfraction$ to sample rfree flags. If we were to instead use something like `DataSet.sample(frac=rfraction)`, we would have a more clean guaranteed bound for how many reflections assigned.

Reply:
Should we leave this as is then?

Reply:
No -- I think we should change this to a more explicit criteria. I'm having a bit of trouble thinking about this test because the threshold also depends on `dmin` (with more reflections, the sampled `rfraction` will be closer to the target).

How about we change the criteria to within 1% of the target number of reflections in the R-free set? This can be expressed as:
```python
x = flags["R-free-flags"].to_numpy()
abs(x.sum() - len(x) * rfraction) / len(x) <= 0.01
```
If done this way, you can also parametrize over `dmin = [1.8, 2.0, 3.0, 5.0]`